1.Faça um algoritmo que:

• leia 20 números inteiros;

• escreva os números que são negativos;

• escreva a média dos números positivos.

In [ ]:
soma_positivos = 0
contador_positivos = 0
contador = 0

while contador < 20:
  numero = int(input('Digite um número inteiro: '))
  if numero < 0:
    print(f'O número {numero} é negativo')
  else:
    soma_positivos += numero
    contador_positivos += 1
  contador += 1

  if contador_positivos > 0:
    media_positivos = soma_positivos / contador_positivos
    print(f'A média dos números positvos é: {media_positivos}')
  else:
    print('Não foram digitados números positivos2')

2.Faça um algoritmo que leia 15 números inteiros e escreva, para cada número lido, se é
par ou ímpar.

In [ ]:
contador = 0

while contador < 15:
  valor = int(input('Informe um número inteiro: '))
  if valor % 2 == 0:
    print(valor, 'é par!')
  else:
    print(valor, 'é ímpar!')
  contador += 1

3.Dado um conjunto de valores inteiros positivos, determinar qual o menor e qual o
maior valor do conjunto. Um número com valor “0” (zero) indica o fim dos dados e
não deve ser considerado.

In [ ]:
menor = float('inf')
maior = float('-inf')

while True:
    valor = int(input("Digite um valor inteiro positivo (0 para sair): "))
    
    if valor == 0:
        break
    
    if valor < menor:
        menor = valor
    
    if valor > maior:
        maior = valor

if menor == float('inf'):
    print("Nenhum valor foi informado")
else:
    print("Menor valor: ", menor)
    print("Maior valor: ", maior)


4.Faça um algoritmo que calcule e escreva a soma dos números pares e a soma dos
números ímpares entre 1 e 100.

In [ ]:
soma_pares = 0
soma_impares = 0

for num in range(1, 101):
  if num % 2 == 0:
    soma_pares += num
  else:
    soma_impares += num
  
print(f'Soma dos números pares entre 1 a 100: {soma_pares}')
print(f'Soma dos números ímpares entre 1 a 100: {soma_impares}')

Soma dos números pares entre 1 a 100: 2550
Soma dos números ímpares entre 1 a 100: 2500


5.Faça um algoritmo que leia a altura de 20 pessoas e calcule a média aritmética das
alturas.

In [ ]:
soma_altura = 0

for i in range(5):
  altura = float(input('Digite sua altura: '))
  soma_altura += altura

media_alturas = soma_altura / 5

print(f'A média aritmética das alturas é {media_alturas}')

Digite sua altura: 120
Digite sua altura: 32
Digite sua altura: 123
Digite sua altura: 231
Digite sua altura: 231
A média aritmética das alturas é 147.4


6.Faça um algoritmo que leia n valores inteiros e escreva quantos desses valores são
negativos.

In [ ]:
n = int(input('Informe a quantidade de números inteiros a serem informados: '))

cont_negativos = 0

for i in range(n):
   valor = int(input("Digite o valor {}: ".format(i+1)))
   if valor < 0:
    cont_negativos += 1

print("Foram encontrados {} valores negativos...".format(cont_negativos))

Informe a quantidade de números inteiros a serem informados: 5
Digite o valor 1: 23
Digite o valor 2: 234
Digite o valor 3: 23
Digite o valor 4: 1
Digite o valor 5: 2
Foram encontrados 0 valores negativos.


7.Faça um algoritmo que leia a quantidade de tinta que uma caneta, e enquanto a
caneta tiver tinta para escrever, escreva “Enquanto tem tinta a caneta escreve...”.
Considere que a cada comando de escrita a caneta gasta 2% da tinta que possui.

In [ ]:
tinta = int(input('Informe a quantidade de tinta da caneta: '))

while tinta > 0:
  print("Enquanto tem tinha a caneta escreve...")
  tinta -= 0.02 * tinta

8.Faça um algoritmo que leia n pares de valores, sendo o primeiro valor o número de
inscrição do atleta e o segundo a altura (em cm) do atleta. Escreva:

• o número de inscrição e a altura do atleta mais alto;

• o número de inscrição e a altura do atleta mais baixo;

• a altura média do grupo de atletas.

In [ ]:
n = int(input('Digite o seu número de inscrição: '))
maior_altura = 0
menor_altura = 0
soma_altura = 0

for i in range(n):
    inscricao = int(input('Digite o seu número de inscrição: '))
    altura = float(input('Digite sua altura: '))
if altura > maior_altura:
    maior_altura = altura
    inscricao_mais_alto = inscricao
if altura < menor_altura:
    menor_altura = altura
    inscricao_mais_baixo = inscricao
soma_altura += altura / n

media_altura = soma_altura / n

print("O atleta mais alto é o número de inscrição", inscricao_mais_alto, "com", maior_altura, "cm.")
print("O atleta mais baixo é o número de inscrição", inscricao_mais_baixo, "com", menor_altura, "cm.")
print("A altura média do grupo de atletas é de", media_altura, "cm.")

9.Faça um algoritmo que calcule e imprima os valores de y, onde:
para x variando de 1.0 até 5.0, em intervalos de 0.1 unidades.

In [ ]:
for x in range(10, 51):
  x /= 10
  y = 3 * x ** 3 - 2 ** x + 7
  print("Para x =", x, ", y =", y)

10.Construir um algoritmo que calcule o fatorial de um número N.

In [ ]:
def fatorial(N):
    resultado = 1
    for i in range(1, N+1):
        resultado *= i
    return resultado

n = 2
print("O fatorial de", n, "é:", fatorial(n))

O fatorial de 2 é: 2


11.Faça um algoritmo que calcule e escreva a soma da seguinte série de 100 termos:

1+ 2 + 3 + 4 +...+100

In [ ]:
i, soma = 0, 0
for i in range(101):
        soma = soma + i
print('\033[1;32m' + f'A soma de 100 (cem) termos é: {soma}')

A soma de 100 (cem) termos é: 5050


12. Faça um algoritmo que calcule e escreva a soma da seguinte série de 100 termos:

In [ ]:
soma = 0
for i in range(1, 101):
  soma += 1 / i
print('\033[1;33m' + f'A soma de 100 (cem) termos é: {soma}')

13. Fazer um algoritmo para calcular o valor de S, dado por:

In [ ]:
N = int(input("digite um numero inteiro positivo: "))
S = 0
den = N
for i in range(1,N+1):
  S= S + i/den
  den= den - 1

  print("S=", S)

digite um numero inteiro positivo: 2
S= 0.5
S= 2.5


14.O valor aproximado do número p pode ser calculado usando-se a série:

sendo o valor de . Faça um algoritmo que calcule e escreva o valor de p
usando os 51 primeiros termos da série.

In [ ]:
n = int(input("Digite o valor de n: "))

p = 0

for i in range(1, n+1):
    p += ((-1) ** (i+1)) / ((2*i-1) ** 2)

p = (8 * p) ** 0.5

print(f"O valor aproximado de pi é {p:.5f}")

15.Fazer um algoritmo que calcule e escreva a soma dos 20 primeiros termos da série:

In [ ]:
soma = 0
for n in range(1, 21):
    termo = (n + 98) / (n**3 + 99)
    soma += termo

print(f"A soma dos 20 primeiros termos da série é {soma:.2f}")

16.Fazer um algoritmo que calcule e imprima o valor de ex através da série:

Considerar para efeitos de cálculo os 30 primeiros termos. O algoritmo deverá ler o
valor de x.

In [ ]:
import math

x = float(input("Digite o valor de x: "))
ex = 0

for n in range(30):
    ex += x**n / math.factorial(n)

print("O valor de e^x é:", ex)


17.Foi feita uma pesquisa de audiência de canal de TV em n casas de um determinado
bairro de Joinville, em um certo dia do mês. Na pesquisa foi utilizado um coletor de
dados portátil. Para cada casa visitada, foi fornecido o número do canal (4, 5, 9, 12) e
o número de pessoas que estavam assistindo a TV naquele horário, considerando que
em cada casa só existia uma televisão. Em casas onde a televisão estava desligada, foi
registrado zero para o número do canal e para o número de pessoas. Faça um
algoritmo que calcule e escreva, para cada emissora, o percentual de audiência.

In [ ]:
emissoras = [4, 5, 9, 12]
total_espectadores = {e: 0 for e in emissoras}

n_casas = int(input("Digite o número de casas visitadas na pesquisa: "))

for i in range(n_casas):
    canal = int(input("Digite o número do canal (4, 5, 9 ou 12): "))
    espectadores = int(input("Digite o número de espectadores: "))
    total_espectadores[canal] += espectadores

for emissora in emissoras:
    percentual = (total_espectadores[emissora] / n_casas) * 100
    print("Emissora:", emissora, " - Percentual de audiência:", percentual, "%")


18.Uma companhia de teatro planeja dar uma série de espetáculos. A direção calcula que,
a 5,00 reais o ingresso, serão vendidos 120 ingressos. Com a diminuição de 0,50 centavos no
preço dos ingressos, espera-se que haja um aumento de 26 ingressos vendidos. As
despesas estão estipuladas em 200,00 reais independente do número de ingressos
vendidos. Faça um algoritmo que escreva uma tabela contendo o preço do ingresso, o
número de ingressos e o lucro esperado em função do preço do ingresso, fazendo-se
variar este preço de 5,00 reais a 1,00 real de 0,50 centavos em 0,50 centavos. Escreva também o lucro
máximo esperado, o preço e o número de ingressos correspondentes.

19.Faça um algoritmo que leia n números inteiros e escreva, para cada número lido, os
divisores e quantidade de divisores.
EXEMPLO: número lido = 12
divisores = 1, 2, 3, 4, 6, 12

quantidade divisores = 6

In [ ]:
n = int(input("Digite a quantidade de números que deseja informar: "))

for i in range(n):
    num = int(input(f"Digite o {i+1}º número: "))
    divisores = []
    cont_divisores = 0
    
    for j in range(1, num+1):
        if num % j == 0:
            divisores.append(j)
            cont_divisores += 1
            
    print(f"Divisores de {num}: {divisores}")
    print(f"Quantidade de divisores de {num}: {cont_divisores}")


20.Uma máquina de biscoito está com problemas. Quando ligada, após 1 hora ela quebra
1 biscoito, na segunda hora ela quebra 3 biscoitos, na hora seguinte ela quebra 3
vezes a quantidade de biscoitos quebrados na hora anterior, e assim por diante. Faça
um algoritmo que calcule quantos biscoitos são quebrados no final de cada dia (a
máquina opera 16 horas por dia).

In [ ]:
tota_biscoitos = 0

for hora in range(1, 17):
  if hora == 1:
    quantidade = 1
  elif hora == 2:
    quantidade = 3
  else:
    total_biscoitos = 3 * quantidade
  
    total_biscoitos += quantidade
    print(f"Horas: {hora} | Biscoitos quebrados: {quantidade}")
    
print(f"Total de biscoitos quebrados em um dia: {total_biscoitos}")


Horas: 3 | Biscoitos quebrados: 3
Horas: 4 | Biscoitos quebrados: 3
Horas: 5 | Biscoitos quebrados: 3
Horas: 6 | Biscoitos quebrados: 3
Horas: 7 | Biscoitos quebrados: 3
Horas: 8 | Biscoitos quebrados: 3
Horas: 9 | Biscoitos quebrados: 3
Horas: 10 | Biscoitos quebrados: 3
Horas: 11 | Biscoitos quebrados: 3
Horas: 12 | Biscoitos quebrados: 3
Horas: 13 | Biscoitos quebrados: 3
Horas: 14 | Biscoitos quebrados: 3
Horas: 15 | Biscoitos quebrados: 3
Horas: 16 | Biscoitos quebrados: 3
Total de biscoitos quebrados em um dia: 12


21.Uma turma tem 50 alunos. Faça um algoritmo que:

• leia para cada aluno o seu nome e idade;

• escreva os nomes dos alunos que tem 18 anos;

• escreva a quantidade de alunos que tem idade acima de 20 anos.

In [ ]:
acima_20 = 0

for i in range(1, 51):
    nome = input(f'Digite o nome do aluno {i}: ')
    idade = int(input(f'Digite a idade do aluno {i}: '))
    
    if idade == 18:
        print(f'O aluno {nome} tem 18 anos.')

    if idade > 20:
        acima_20 += 1

print(f'Existem {acima_20} alunos com idade acima de 20 anos.')

22.Faça um algoritmo que:

• leia, para n pessoas, a altura e o sexo (sexo = 'M' ou sexo = 'm' para masculino e
sexo = 'F' ou sexo = 'f' para feminino);

• escreva a média da altura das mulheres;

• escreva a média da altura da turma.

In [ ]:
n = int(input('Digite o número de pessoas: '))

soma_alturas_mulheres = 0
qtd_mulheres = 0
soma_altura_turma = 0

for i in range(n):
  altura = float(input('Informe a sua altura em metros: '))
  sexo = input('Digite o sexo (M/F): ')

  soma_altura_turma += altura

  if sexo.upper() == 'F':
        soma_alturas_mulheres += altura
        qtd_mulheres += 1

media_altura_mulheres = soma_alturas_mulheres / qtd_mulheres if qtd_mulheres > 0 else 0
media_altura_turma = soma_altura_turma / n if n > 0 else 0

print(f"Média da altura das mulheres: {media_altura_mulheres:.2f} m")
print(f"Média da altura da turma: {media_altura_turma:.2f} m")

23.Uma loja de departamentos oferece para seus clientes um determinado desconto de
acordo com o valor da compra efetuada. O desconto é de 20% caso o valor da compra
seja maior que R$ 500,00 e de 15% caso seja menor ou igual. Faça um algoritmo que
leia, para cada cliente, nome, endereço e valor da compra e escreva o total a pagar.
Um nome de cliente igual a ULTIMO indica o fim da entrada de dados.

In [ ]:
desconto_maior = 0.2
desconto_menor = 0.15
total_vendas = 0

while True:
  nome = input('Digite o nome do cliente: ')
  if nome.upper() == 'ÚLTIMO':
    break
  endereco = input('Digite o endereço do cliente: ')
  valor_compra = float(input('Digite o valor da compra: '))

  if valor_compra > 500:
    desconto = desconto_maior
  else:
    desconto = desconto_menor
  valor_desconto = valor_compra * desconto
  valor_pagar = valor_compra - valor_desconto

  print(f'Cliente: {nome}')
  print(f'Endereço: {endereco}')
  print(f'Valor da compra: R${valor_compra:.2f}')
  print(f'Desconto: {desconto*100:.0f}%')
  print(f'Valor do desconto: R${valor_desconto:.2f}')
  print(f'Total a pagar: R${valor_pagar:.2f}')
  print("-" * 30)

  total_vendas += valor_pagar

print(f'Total de vendas: R${total_vendas:.2f}')

24.Faça um algoritmo que leia valores, sendo que cada valor representa a idade de uma
pessoa. Calcule e escreva a idade média do grupo de pessoas. Só devem ser
computados no cálculo valores maiores do que zero. O algoritmo deve apresentar ao
usuário a seguinte mensagem:

deseja digitar mais um valor: s (SIM) / n (NAO)?,
antes de prosseguir com a entrada de dados.

In [ ]:
idades = []
mais_um_valor = 's'

while mais_um_valor == 's':
    idade = int(input("Digite a idade: "))
    if idade > 0:
        idades.append(idade)
    mais_um_valor = input("Deseja digitar mais um valor? (s/n)").lower()

if len(idades) > 0:
    media_idades = sum(idades) / len(idades)
    print(f"A média das idades é {media_idades:.2f}")
else:
    print("Não foram digitadas idades válidas.")

25.Um hotel cobra 50,00 reais de diária por hóspede e mais uma taxa de serviços. A taxa de
serviços é de:

• 7,50 reais por diária, caso o número de diárias seja menor que 15;

• 6,50 reais por diária, caso o número de diárias seja igual a 15;

• 5,00 reais por diária, caso o número de diárias seja maior que 15.

Faça um algoritmo que apresente as seguintes opções ao recepcionista:
1. encerrar a conta de um hóspede
2. verificar número de contas encerradas
3. finalizar a execução
Caso a opção escolhida seja a primeira, leia o nome e o número de diárias do hóspede
e escreva o nome e total a ser pago. Caso a opção escolhida seja a segunda, informe o
número de hóspedes que deixaram o hotel (número de contas encerradas).

In [ ]:
valor_diaria = 50.0
taxa_servico_menor15 = 7.5
taxa_servico_igual15 = 6.5
taxa_servico_maior15 = 5.0

contas_encerradas = 0
hospedes_encerrados = []

while True:
    print("\nSelecione uma opção:")
    print("1 - Encerrar conta de um hóspede")
    print("2 - Verificar número de contas encerradas")
    print("3 - Finalizar execução")

    opcao = int(input("Digite sua escolha (1, 2 ou 3): "))

    if opcao == 1:
        nome = input("Digite o nome do hóspede: ")
        diarias = int(input("Digite o número de diárias: "))

        total_diarias = valor_diaria * diarias

        if diarias < 15:
            taxa_servico = taxa_servico_menor15
        elif diarias == 15:
            taxa_servico = taxa_servico_igual15
        else:
            taxa_servico = taxa_servico_maior15

        total_servico = taxa_servico * diarias
        total_conta = total_diarias + total_servico

        print(f"\nNome: {nome}")
        print(f"Total a pagar: R${total_conta:.2f}")
        hospedes_encerrados.append(nome)
        contas_encerradas += 1

    elif opcao == 2:
        print(f"\nNúmero de contas encerradas: {contas_encerradas}")
        print(f"Hóspedes encerrados: {hospedes_encerrados}")

    elif opcao == 3:
        break

    else:
        print("Opção inválida! Tente novamente.")

26.Um determinado material radioativo perde metade de sua massa a cada 50 segundos.
Dada a sua massa inicial em Kg, faça um algoritmo que determine o tempo necessário
para que essa massa se torne menor que 0,5 gramas. Escreva a massa inicial, a massa
final e o tempo.

In [13]:
import math

massa_inicial = float(input('Digite a massa incial (kg): '))
massa_final =  0.0005
t = 50 * math.log2(massa_inicial/massa_final)

print(f'Massa inicial: {massa_inicial} kg')
print(f'Massa final: {massa_final} kg')
print(f'Tempo necessário: {t:.2f} segundos')

Digite a massa incial (kg): 20
Massa inicial: 20.0 kg
Massa final: 0.0005 kg
Tempo necessário: 764.39 segundos


27.Um motorista acaba de voltar de um feriado prolongado. Antes de sair de viagem e
imediatamente após retornar, o motorista encheu o tanque do veículo e registrou as
medidas do odômetro. Em cada parada feita durante a viagem, foi registrado o valor
do odômetro e a quantidade de combustível comprado para reabastecer o veículo
(suponha que o tanque ficou vazio e foi enchido a cada parada). Faça um algoritmo
que leia o número total de reabastecimentos feitos (incluindo o primeiro) e os dados
registrados relativos à compra de combustível. Calcule e escreva:

a) a quilometragem obtida por litro de combustível entre cada par de paradas

b) a quilometragem média obtida por litro de combustível em toda a viagem.

In [ ]:
n = int(input("Número de reabastecimentos feitos: "))

odometro_anterior = int(input("Odômetro inicial: "))
quantidade_combustivel = float(input("Quantidade de combustível comprado (em litros): "))

distancia_total = 0
combustivel_total = quantidade_combustivel

for i in range(2, n+1):
    odometro_atual = int(input(f"Odômetro na parada {i}: "))
    quantidade_combustivel = float(input(f"Quantidade de combustível comprado na parada {i} (em litros): "))

    distancia_parcial = odometro_atual - odometro_anterior
    quilometragem = distancia_parcial / quantidade_combustivel

    print(f"Quilometragem obtida entre a parada {i-1} e a parada {i}: {quilometragem:.2f} km/l")

    distancia_total += distancia_parcial
    combustivel_total += quantidade_combustivel
    odometro_anterior = odometro_atual

quilometragem_media = distancia_total / combustivel_total
print(f"Quilometragem média em toda a viagem: {quilometragem_media:.2f} km/l")


28.Em uma disputa de pingue-pongue os pontos são anotados como D, ponto para o
jogador do lado direito, e E, ponto para o jogador do lado esquerdo da mesa. Faça um
algoritmo que leia o código do ponto de cada jogada e determine o vencedor. A
partida encerra quando:

a) um dos jogadores chegar a 21 pontos e a diferença de pontos entre os jogadores for
maior ou igual a dois;

b) o jogador com mais de 21 pontos conseguir uma diferença de dois pontos sobre o
adversário, caso a primeira condição não seja atendida.

In [ ]:
direita = 0
esquerda = 0
vencedor = None

while vencedor is None:
  ponto = input('Digite o código do ponto (D/E): ')
  if ponto == 'D':
    direita += 1
  elif ponto == 'E':
    esquerda += 1
  else:
    print('\033[1;31m' + 'Ponto inválido! Tente novamente.')
    continue

    diferenca = abs(direita - esquerda)
    if (direita >= 21 and diferenca >= 2) or (esquerda >= 21 and diferenca >= 2):
        if direita > esquerda:
            vencedor = "direita"
        else:
            vencedor = "esquerda"
    elif pontos_direita > 21 and direita > esquerda + 1:
        vencedor = "direita"
    elif esquerda > 21 and esquerda > direita + 1:
        vencedor = "esquerda"

print('\033[1;33m' + f"Partida encerrada. O vencedor é o jogador da {vencedor}!")

29.Os regulamentos de uma competição de pesca impõem um limite no peso total de
pesca de um dia. Faça um algoritmo que leia o limite diário (em quilogramas) e então
leia o peso (em gramas) de cada peixe e escreva o peso total da pesca obtido até
aquele ponto. Quando o limite diário for excedido escreva uma mensagem e encerre a
execução do algoritmo. O algoritmo deve ainda apresentar ao usuário a seguinte
mensagem: informar o peso de mais um peixe: s (SIM) / n (NÃO)? antes de prosseguir
com a entrada de dados.

In [ ]:
limite = float(input('Digite o limite diário de pesca (em kg): '))
peso_total = 0
sim = 's'

while sim == 's':
  peso = float(input('Informe o peso do peixe (em gramas): '))
  peso_total += peso / 1000

  if peso_total > limite:
    print('\033[1;31m' + 'Limite excedido!')
    break

  sim = input('Deseja informar o peso de mais um peixe? (s/n)').lower()

print(f'Peso total: {peso_total:.2f} kg')


30.Foi feita uma pesquisa do consumo mensal de energia elétrica em uma determinada
cidade. Para cada consumidor, são fornecidos os seguintes dados: número de
identificação do consumidor, quantidade de kWh consumidos durante o mês, código
do tipo de consumidor (R - residencial, C - comercial, I - industrial). Faça um algoritmo
que:

a) leia o preço do kWh por tipo de consumidor;

b) leia os dados de n consumidores;

c) escreva o número de identificação e o total a pagar, para cada consumidor;

d) escreva a quantidade total de KWh consumida para cada um dos três tipos de
consumidores;

e) escreva a quantidade média geral de consumo.

In [ ]:
preco_residencial = float(input('Digite o preço residencial do kWh para consumidores residenciais: '))
preco_comercial = float(input('Digite o preço comercial do kWh para consumidores residenciais: '))
preco_industrial = float(input('Digite o preço industrial do kWh para consumidores residenciais: '))

total_residencial = total_comercial = total_industrial = quantidade_total = 0

n = int(input('Digite o número de consumidores: '))

for i in range(n):
  identificacao = int(input('Digite o número de identificação do consumidor: '))
  quantidade_kwh = float(input("Digite a quantidade de kWh consumida: "))
  tipo_consumidor = input("Digite o tipo do consumidor (R/C/I): ")

  if tipo_consumidor == "R":
        valor_pagar = quantidade_kwh * preco_residencial
        total_residencial += quantidade_kwh
  elif tipo_consumidor == "C":
        valor_pagar = quantidade_kwh * preco_comercial
        total_comercial += quantidade_kwh
  elif tipo_consumidor == "I":
        valor_pagar = quantidade_kwh * preco_industrial
        total_industrial += quantidade_kwh
  else:
        print("Tipo de consumidor inválido.")
        continue

  print(f"Consumidor {identificacao}: total a pagar = R$ {valor_pagar:.2f}")
  quantidade_total += quantidade_kwh

print(f"Total de kWh consumidos por residenciais: {total_residencial}")
print(f"Total de kWh consumidos por comerciais: {total_comercial}")
print(f"Total de kWh consumidos por industriais: {total_industrial}")

quantidade_media = quantidade_total / n
print(f"Quantidade média geral de consumo: {quantidade_media:.2f} kWh")